config/src/main/java/com/typesafe/config/ConfigFactory.java

```diff
@@ -1090,6 +1090,63 @@ public static Config parseResourcesAnySyntax(String resourceBasename) {
         return parseResourcesAnySyntax(resourceBasename, ConfigParseOptions.defaults());
     }
 
+    /**
+     * Parse only any application overrides (those specified by config.{resource,file,url}), returning
+     * an empty Config if no overrides were set.
+     * @param parseOptions parse options
+     * @return the parsed configuration
+     */
+    public static Config parseApplicationOverride(ConfigParseOptions parseOptions) {
+        ClassLoader loader = parseOptions.getClassLoader();
+
+        if (loader == null)
+            throw new ConfigException.BugOrBroken(
+                "ClassLoader should have been set here; bug in ConfigFactory. "
+                    + "(You can probably work around this bug by passing in a class loader or calling currentThread().setContextClassLoader() though.)");
+
+        int specified = 0;
+
+        // override application.conf with config.file, config.resource,
+        // config.url if requested.
+        String resource = System.getProperty("config.resource");
+        if (resource != null)
+            specified += 1;
+        String file = System.getProperty("config.file");
+        if (file != null)
+            specified += 1;
+        String url = System.getProperty("config.url");
+        if (url != null)
+            specified += 1;
+
+        if (specified == 0) {
+            return ConfigImpl.emptyConfig("TODO: what to put here? Should something else be returned?");
```

Comment:
maybe use Java 8 Optional ... ? I don't know what current best practice is around that, I haven't Java'd in a few years. I think there is a difference here between "something was specified and it was empty" and "nothing was specified."

Reply:
This sounds good.  Other option would be to return null I guess, but the way `Optional` allows `orElseGet` in `DefaultConfigLoadingStrategy` is nice, so will use `Optional`.